common/web/types/src/util/util.ts

```diff
@@ -101,3 +101,152 @@ toOneChar(value: string) : number {
   }
   return value.codePointAt(0);
 }
+
+export function describeCodepoint(ch : number) : string {
```

Comment:
I see some similarities in `getProblem` (ll 215-221).
Maybe could be refactored like
```
let s = getProblem(ch);
if (s == null) {
  s = Strring.fromCodePoint(ch);
}
return `"${s}" (U+${Number(ch).toString(16)})`;
```

(additional changes for the casing difference from BadStringType)

Reply:
good point!